src/internal/Core.sol

```diff
@@ -431,10 +432,39 @@ library Core {
         if (opts.defender.useDefenderDeploy) {
             return DefenderDeploy.deploy(contractName, constructorData, opts.defender);
         } else {
-            return _deploy(contractName, constructorData);
+            if (opts.defender.salt != 0) {
```

Comment:
can I suggest that we explicitly create a new option here with a different name? (e.g.: `opts.customSalt`)

reason being that we want the script to actually fail if we accidentally run it against the original code instead of the fork.
If I'm not mistaken, by reusing `defender.salt`, that scenario would silently use the original behaviour with no warning?

Reply:
Ok, let me add it